bftengine/src/bftengine/messages/ClientRequestMsg.cpp

```diff
@@ -32,12 +32,17 @@ uint32_t getRequestSizeTemp(const char* request)  // TODO(GG): change - TBD
 }
 
 // class ClientRequestMsg
-
-ClientRequestMsg::ClientRequestMsg(
-    NodeIdType sender, uint8_t flags, uint64_t reqSeqNum, uint32_t requestLength, const char* request)
-    : MessageBase(sender, MsgCode::ClientRequest, (sizeof(ClientRequestMsgHeader) + requestLength)) {
+ClientRequestMsg::ClientRequestMsg(NodeIdType sender,
+                                   uint8_t flags,
+                                   uint64_t reqSeqNum,
+                                   uint32_t requestLength,
+                                   const char* request,
+                                   const std::string& cid)
+    : MessageBase(sender, MsgCode::ClientRequest, (sizeof(ClientRequestMsgHeader) + requestLength + cid.size() + 1)) {
   setParams(sender, reqSeqNum, requestLength, flags);
+  msgBody()->cid_length = cid.size() + 1;
   memcpy(body() + sizeof(ClientRequestMsgHeader), request, requestLength);
+  strcpy(body() + sizeof(ClientRequestMsgHeader) + requestLength, cid.c_str());
```

Comment:
Please use `memcpy` instead. At a minimum, use `strncpy` if we are sure the string doesn't actually contain null characters. Raw `strcpy` is a vector for buffer overflow attacks.

Reply:
FIxed